tests/unit/class/xml/rpc/bloggerapiTest.php

```diff
@@ -118,21 +118,117 @@ function test_getRecentPosts()
 
     function test_getUsersBlogs()
     {
-		$this->markTestIncomplete();
+		$params = array(null, 'admin', 'dummy');
+		$response = new XoopsXmlRpcResponse();
+		$module = new XoopsModule();
+		$instance = new $this->myClass($params, $response, $module);
+        
+        $result = $instance->getUsersBlogs();
+        $msg = $response->render();
+        $expected = '<?xml version="1.0"?>'
+            . '<methodResponse><fault><value><struct>'
+            . '<member><name>faultCode</name><value>104</value></member>'
+            . "<member><name>faultString</name><value>User authentication failed\n</value></member>"
+            . '</struct></value></fault></methodResponse>';
+        $this->assertSame($expected, $msg);
+        
+        
+		$params = array(null, 'admin', 'adminadmin');
+		$response = new XoopsXmlRpcResponse();
+		$module = new XoopsModule();
+		$instance = new $this->myClass($params, $response, $module);
+        
+        $result = $instance->getUsersBlogs();
+        $msg = $response->render();
+        $expected = '<?xml version="1.0"?><methodResponse><params><param><value><array><data><value><struct>'
+            . '<member><name>url</name><value><string>http://localhost/projects/www/XoopsCore/htdocs/modules//</string></value></member>'
+            . '<member><name>blogid</name><value><string></string></value></member>'
+            . '<member><name>blogName</name><value><string>XOOPS Blog</string></value></member>'
+            . '</struct></value></data></array></value></param></params></methodResponse>';
+        $this->assertSame($expected, $msg);
+        
     }
 
     function test_getUserInfo()
     {
-		$this->markTestIncomplete();
+		$params = array(null, 'admin', 'dummy');
```

Comment:
Can we mock the XoopsUser used for this somehow? The particulars (name, password, etc.) are different in every environment. That results in failures.

Just a quick glance, but maybe just extend BloggerApi and override the user handling?
```
class MockBloggerApi extends \BloggerApi
{
    function _checkUser($username, $password)
    {
        $xoops = Xoops::getInstance();

        $member_handler = $xoops->getHandlerMember();
        $userObject = new \Xoops\Core\Kernel\Handlers\XoopsUser();
        $this->user = null;
        $this->admin = false;
        if ($username == 'admin' && $password == 'goodpassword') {
            $userObject['uid'] = 1;
            $userObject['name'] = ucfirst($username);
            $userObject['uname'] = $username;
            // etc.
            $this->user = $userObject;
            $this->admin = true;
            return true;
        } elseif ($username == 'reguser' && $password == 'goodpassword') {
            $userObject['uid'] = 99999;
            $userObject['name'] = ucfirst($username);
            $userObject['uname'] = $username;
            // etc.
            $this->user = $userObject;
            return true;
        } // etc...
        return false;
    }
}
```

Something like that would make the test results independent of the actual environment.